### PR TITLE
Mui 5 Migration of Some components

### PR DESCRIPTION
**Pull request by abhijeetChawla**
Migrated some components to the mui 5 system

### CODE REVIEW

File: src/components/ExtensionCard.tsx [hunk 1]
```diff
@@ -6,42 +6,30 @@
...
-    },
-    card: {
-        margin: '10px',
+const CardRoot = styled(CardContent)({
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    padding: 16,
+    '&:last-child': {
+        paddingBottom: '16px',
```
Comment (ntbm) [inline src/components/ExtensionCard.tsx:24]: That wasn't here before, why did you add it?

File: src/components/ExtensionCard.tsx [hunk 2]
```diff
@@ -6,42 +6,30 @@
...
     },
+});
+
+const CardIcon = styled(Avatar)(({ theme }) => ({
+    width: theme.spacing(8),
+    height: theme.spacing(8),
```
Comment (ntbm) [inline src/components/ExtensionCard.tsx:30]: You're changing the spacing here as well

File: src/components/SourceCard.tsx [hunk 3]
```diff
@@ -18,56 +17,33 @@ import Avatar from '@mui/material/Avatar';
...
-        justifyContent: 'space-between',
-        alignItems: 'center',
-        padding: 16,
+const CardRoot = styled(CardContent)({
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    padding: '16px',
+    '&:last-child': {
+        paddingBottom: '16px',
```
Comment (ntbm) [inline src/components/SourceCard.tsx:28]: Wasn't here before

File: src/components/SourceCard.tsx [hunk 4]
```diff
@@ -18,56 +17,33 @@ import Avatar from '@mui/material/Avatar';
...
+    margin: '10px',
+    '&:hover,&:active': {
+        backgroundColor: theme.palette.action.selected,
+        transition: 'background-color 100ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
     },
 }));
 
+const CardIcon = styled(Avatar)(({ theme }) => ({
+    width: theme.spacing(8),
+    height: theme.spacing(8),
```
Comment (ntbm) [inline src/components/SourceCard.tsx:42]: different spacing

File: src/screens/settings/About.tsx [hunk 5]
```diff
@@ -36,7 +24,7 @@ export default function About() {
 
     if (about === undefined) {
         return (
-            <div className={classes.loading}>
+            <div style={{ display: 'grid', placeItems: 'center', height: '80vh' }}>
```
Comment (ntbm) [inline src/screens/settings/About.tsx:27]: Need to check if this is visually correct

File: src/screens/Reader.tsx [hunk 6]
```diff
@@ -145,7 +130,10 @@ export default function Reader() {
     // return spinner while chpater data is loading
     if (chapter.pageCount === -1) {
         return (
-            <div className={classes.loading}>
+            <div style={{
+                display: 'grid', placeItems: 'center', width: '100vw', height: '100vh',
+            }}
+            >
```
Comment (ntbm) [inline src/screens/Reader.tsx:136]: Need to check if this is visually correct

File: src/App.tsx [hunk 7]
```diff
@@ -105,10 +103,10 @@ export default function App() {
...
                                 maxWidth={false}
                                 disableGutters
-                                style={{
-                                    marginTop: theme.spacing(8),
-                                    marginLeft: isMobileWidth ? '' : theme.spacing(8),
-                                    marginBottom: isMobileWidth ? theme.spacing(8) : '',
+                                sx={{
+                                    marginTop: 8,
+                                    marginLeft: { sm: 8 },
+                                    marginBottom: { xs: 8, sm: 4 },
```
Comment (AriaMoradi) [inline src/App.tsx:109]: keys like `mt`, `ml`, `mb` exist too

File: src/components/ExtensionCard.tsx (hunk 1 above)
Comment (AriaMoradi) [inline src/components/ExtensionCard.tsx:24]: `pb: 2`

File: src/components/ExtensionCard.tsx [hunk 8]
```diff
@@ -118,16 +104,15 @@ export default function ExtensionCard(props: IProps) {
     }
 
     return (
-        <Card className={classes.card}>
-            <CardContent className={classes.root}>
-                <div style={{ display: 'flex' }}>
-                    <Avatar
+        <Card style={{ margin: '10px' }}>
```
Comment (AriaMoradi) [inline src/components/ExtensionCard.tsx:107]: use `sx` instead

File: src/components/ExtensionCard.tsx [hunk 9]
```diff
@@ -118,16 +104,15 @@ export default function ExtensionCard(props: IProps) {
...
+            <CardRoot>
+                <Box display="flex">
```
Comment (AriaMoradi) [inline src/components/ExtensionCard.tsx:109]: use `sx` for css properties

File: src/components/ExtensionCard.tsx [hunk 10]
```diff
@@ -118,16 +104,15 @@ export default function ExtensionCard(props: IProps) {
...
+                    <CardIcon
                         variant="rounded"
-                        className={classes.icon}
                         alt={name}
                         src={`${serverAddress}${iconUrl}?useCache=${useCache}`}
                     />
-                    <div style={{ display: 'flex', flexDirection: 'column' }}>
+                    <Box display="flex" flexDirection="column">
```
Comment (AriaMoradi) [inline src/components/ExtensionCard.tsx:115]: use `sx`

File: src/screens/Reader.tsx [hunk 11]
```diff
@@ -176,7 +164,7 @@ export default function Reader() {
     const ReaderComponent = getReaderComponent(settings.readerType);
 
     return (
-        <div className={classes.root}>
+        <div style={{ width: settings.staticNav ? 'calc(100vw - 300px)' : '100vw' }}>
```
Comment (AriaMoradi) [inline src/screens/Reader.tsx:167]: `sx`

File: src/screens/Reader.tsx [hunk 12]
```diff
@@ -145,7 +130,10 @@ export default function Reader() {
     // return spinner while chpater data is loading
     if (chapter.pageCount === -1) {
         return (
-            <div className={classes.loading}>
+            <div style={{
+                display: 'grid', placeItems: 'center', width: '100vw', height: '100vh',
+            }}
```
Comment (AriaMoradi) [inline src/screens/Reader.tsx:135]: `sx`

**Review by AriaMoradi — CHANGES_REQUESTED** [review]

File: src/components/navbar/ReaderNavBar.tsx [hunk 13]
```diff
@@ -28,113 +28,102 @@ import ListItemSecondaryAction from '@mui/material/ListItemSecondaryAction';
 import Collapse from '@mui/material/Collapse';
 import Button from '@mui/material/Button';
 import NavBarContext from 'components/context/NavbarContext';
+import { styled } from '@mui/system';
 
-const useStyles = (settings: IReaderSettings) => makeStyles((theme) => ({
+const useStyles = makeStyles({
```
Comment (AriaMoradi) [inline src/components/navbar/ReaderNavBar.tsx:33]: There should be a way to eliminate this?

**Review by AriaMoradi — CHANGES_REQUESTED** [review]

**Comment by abhijeetChawla** [comment]
So finally done, migration to the MUI 5 system is almost done. Only the manga screen component is left to converted and for that i want to make a seperate pull request

File: src/components/navbar/ReaderNavBar.tsx [hunk 14]
```diff
@@ -173,15 +160,15 @@ export default function ReaderNavBar(props: IProps) {
...
-        rootEl.classList.add(classes.AppRootElment);
-        mainContainer.classList.add(classes.AppMainContainer);
+        rootEl.style.display = 'flex';
+        mainContainer.style.display = 'none';
 
         return () => {
-            rootEl.classList.remove(classes.AppRootElment);
-            mainContainer.classList.remove(classes.AppMainContainer);
+            rootEl.style.display = 'block';
+            mainContainer.style.display = 'block';
```
Comment (AriaMoradi) [inline src/components/navbar/ReaderNavBar.tsx:171]: don't remove the comment

**Review by AriaMoradi — CHANGES_REQUESTED** [review]

**Review by AriaMoradi — APPROVED** [review]

**Comment by AriaMoradi** [comment]
@ntbm Can I also have your approval? This is a big PR and I might have missed things in my review.

**Comment by ntbm** [comment]
I’m out and about right now. Will look at it today
Aria Moradi ***@***.***> schrieb am Sa. 13. Nov. 2021 um
17:32:

**Comment by ntbm** [comment]
Sources is broken
Bildschirmfoto 2021-11-13 um 21 04 23
Bildschirmfoto 2021-11-13 um 21 04 15

File: src/components/SourceCard.tsx [hunk 15]
```diff
@@ -140,32 +104,32 @@ export default function SourceCard(props: IProps) {
...
+                            sx={{ ml: 20 }}
                             onClick={(e) => redirectTo(e, `/sources/${id}/search/`)}
                         >
                             Search
                         </Button>
                         {supportsLatest && (
                             <Button
                                 variant="outlined"
-                                style={{ marginLeft: 20 }}
+                                sx={{ ml: 20 }}
```
Comment (ntbm) [inline src/components/SourceCard.tsx:119]: marginLeft: 20 is 20px ml: 20 is theme.spacing(20)

File: src/components/SourceCard.tsx [hunk 16]
```diff
@@ -140,32 +104,32 @@ export default function SourceCard(props: IProps) {
...
                             </IconButton>
                         )}
-                    </div>
-                    <div className={classes.showBigger}>
+                    </Box>
+                    <Box sx={{ display: { xs: 'none', sm: 'flex' } }}>
                         <Button
                             variant="outlined"
-                            style={{ marginLeft: 20 }}
+                            sx={{ ml: 20 }}
```
Comment (ntbm) [inline src/components/SourceCard.tsx:111]: marginLeft: 20 is 20px ml: 20 is theme.spacing(20)

File: src/components/SourceCard.tsx [hunk 17]
```diff
@@ -140,32 +104,32 @@ export default function SourceCard(props: IProps) {
...
+                                sx={{ ml: 20 }}
                                 onClick={(e) => redirectTo(e, `/sources/${id}/latest/`)}
                             >
                                 Latest
                             </Button>
                         )}
                         <Button
                             variant="outlined"
-                            style={{ marginLeft: 20 }}
+                            sx={{ ml: 20 }}
```
Comment (ntbm) [inline src/components/SourceCard.tsx:127]: marginLeft: 20 is 20px ml: 20 is theme.spacing(20)

**Comment by AriaMoradi** [comment]
Also fix the merge conflict

File: src/components/ChapterCard.tsx [hunk 18]
```diff
@@ -101,8 +102,8 @@ export default function ChapterCard(props: IProps) {
                                 alignItems: 'center',
                                 padding: 2,
                             }}
-                        >
-                            <div style={{ display: 'flex' }}>
+                        >
```
Comment (ntbm) [inline src/components/ChapterCard.tsx:105]: Remove trailing whitespace
```suggestion
                        >
```

**Review by ntbm — CHANGES_REQUESTED** [review]

**Review by ntbm — APPROVED** [review]

**Pull request merged**